internal/workerutil/worker.go

```diff
@@ -147,12 +172,15 @@ loop:
 		}
 
 		select {
-		case <-w.clock.After(delay):
+		case <-w.dequeueClock.After(delay):
 		case <-w.ctx.Done():
 			break loop
+		case <-shutdownChan:
+			break loop
```

Comment:
Maybe add a log here to indicate the _reason_ it's shutting down? 
```suggestion
		case <-shutdownChan:
		    log15.Info("MaxActiveTime elapsed, stopping dequeue loop")
			break loop
```